Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/azure/serverless/AzureSparkCosmosClusterManager.java

```diff
@@ -261,6 +255,13 @@ public ClusterContainer refresh() {
                 .doOnNext(triples -> log().debug("Triple(Subscription, AccountBasic, AccountDetails) list: " + triples.toString()));
     }
 
+    public Observable<DataLakeAnalyticsAccount> getAzureDataLakeAccountDetail(AzureHttpObservable http, String adlaResouceId) {
```

Comment:
`http` has to be annotated `@Nullable`

Reply:
There is a method call `getAzureDataLakeAccountDetail(getHttp(), getAdlaResourceId())`  at class `SynapseCosmosSparkPool`, and `getHttp()` method is a Nullable method. But if we check the context at the method call, we will find variable `http` is already set with a NotNull value before we call `getAzureDataLakeAccountDetail`. Therefore, I won't add annotation `Nullable` here since it's by design NotNull. Instead, I will replace `getAzureDataLakeAccountDetail(getHttp(), getAdlaResourceId())` with  `getAzureDataLakeAccountDetail(this.http, getAdlaResourceId())` to make it easier to understand.

---
In reply to: [378084241](https://github.com/microsoft/azure-tools-for-java/pull/3982#discussion_r378084241) [](ancestors = 378084241)